api/locations/index.ts

```diff
@@ -0,0 +1,56 @@
+import { NextFunction, Response } from 'express';
+import { handleGet } from '../common/mockService';
+import { getConfigValue } from '../configuration';
+import { SERVICES_PRD_API_URL} from '../configuration/references';
+import { EnhancedRequest } from '../lib/models';
+import * as mock from '../locations/location.mock';
+import {LocationTypeEnum} from './data/locationType.enum';
+import {SERVICES_COURT_TYPE_MAPPINGS} from './data/serviceCourtType.mapping';
+import {LocationModel} from './models/location.model';
+
+mock.init();
+
+const url: string = getConfigValue(SERVICES_PRD_API_URL);
+
+/**
+ * @description getLocations from service ID/location type/search term
+ * @overview API sample: /api/locations/getLocations?serviceIds=SSCS,IA&locationType=hearing&searchTerm=CT91RL
+ * @example service = SSCS | SSCS,IA split with ','
+ * @example locationType = optional | hearing | case_management
+ * @example searchTerm = any search term for postcode | site name | venue name |court name | court address etc.
+ */
+export async function getLocations(req: EnhancedRequest, res: Response, next: NextFunction) {
+  // @ts-ignore
+  const searchTerm = req.query.searchTerm;
```

Comment:
you could "spread" this so it's a concise one line:
`const { searchTerm, serviceIds, locationType } = req.query;`

Reply:
Merged from https://github.com/hmcts/rpx-xui-webapp/tree/feature/EUI-4690-location-api-to-master - not part of the changes in this PR.